v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,943 @@
+    try {
+      ZonedDateTime dateTime = ZonedDateTime.parse(value);
+      ZoneOffset offset = ZoneOffset.of(timezoneOffset);
+      dateTime = dateTime.withZoneSameInstant(offset);
+      return "'" + dateTime.withZoneSameInstant(ZoneOffset.UTC).toString() + "'";
+    } catch (DateTimeParseException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Converts a string representation of a date to a {@link LocalDate} compatible with Cassandra.
+   *
+   * <p>The method parses the {@code dateString} into an {@link Instant}, converts it to a {@link
+   * Date}, and then retrieves the corresponding {@link LocalDate} from the system's default time
+   * zone.
+   *
+   * @param dateString The date string in ISO-8601 format (e.g., "2024-12-05T00:00:00Z").
+   * @return The {@link LocalDate} representation of the date.
+   */
+  public static LocalDate convertToCassandraDate(String dateString) {
+    Instant instant = Instant.parse(dateString);
+    Date date = Date.from(instant);
+    return date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+  }
+
+  /**
+   * Converts a string representation of a timestamp to an {@link Instant} compatible with
+   * Cassandra.
+   *
+   * <p>The method parses the {@code dateString} into an {@link Instant}, which represents an
+   * instantaneous point in time and is compatible with Cassandra timestamp types.
+   *
+   * @param dateString The timestamp string in ISO-8601 format (e.g., "2024-12-05T10:15:30Z").
+   * @return The {@link Instant} representation of the timestamp.
+   */
+  public static Instant convertToCassandraTimestamp(String dateString) {
+    return Instant.parse(dateString);
+  }
+
+  /**
+   * Validates if the given string represents a valid UUID.
+   *
+   * <p>This method attempts to parse the provided string as a UUID using {@link
+   * UUID#fromString(String)}. If parsing is successful, it returns {@code true}, indicating that
+   * the string is a valid UUID. Otherwise, it returns {@code false}.
+   *
+   * @param value The string to check if it represents a valid UUID.
+   * @return {@code true} if the string is a valid UUID, {@code false} otherwise.
+   */
+  public static boolean isValidUUID(String value) {
+    try {
+      UUID.fromString(value);
+      return true;
+    } catch (IllegalArgumentException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Validates if the given string represents a valid IP address.
+   *
+   * <p>This method attempts to resolve the provided string as an {@link InetAddress} using {@link
+   * InetAddress#getByName(String)}. If successful, it returns {@code true}, indicating that the
+   * string is a valid IP address. Otherwise, it returns {@code false}.
+   *
+   * @param value The string to check if it represents a valid IP address.
+   * @return {@code true} if the string is a valid IP address, {@code false} otherwise.
+   */
+  public static boolean isValidIPAddress(String value) {
+    try {
```

Comment:
This involves a DNS query, let's avoid this by using
[com.google.common.net.InetAddresses#forString(lava.lang.String)](https://guava.dev/releases/20.0/api/docs/com/google/common/net/InetAddresses.html#forString-java.lang.String-)

Reply:
Address the same @VardhanThigle

Reply:
Thanks. As this has migrated to the `com.google.common.net.InetAddresses#forString` it's fine with me.
I am assuming in the larger picture you are calling this to validate. Resolving.